### PR TITLE
[NativeAOT] Fixes resource warning on a HW app

### LINKED ISSUE #25740: PublishTrimmed is set too late when AOT publishing
```
dotnet new console
dotnet publish -r win-x64 -c Release /p:PublishAot=true
```

Running the above will produce following warning:

```
/_/src/libraries/System.Private.CoreLib/src/System/Resources/ManifestBasedResourceGroveler.cs(239): Trim analysis warni
ng IL2026: System.Resources.ManifestBasedResourceGroveler.CreateResourceSet(Stream,Assembly): Using member 'System.Reso
urces.ManifestBasedResourceGroveler.InternalGetResourceSetFromSerializedData(Stream,String,String,ResourceManagerMediat
or)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. The CustomRe
sourceTypesSupport feature switch has been enabled for this app which is being trimmed. Custom readers as well as custo
m objects on the resources file are not observable by the trimmer and so required assemblies, types and members may be
removed. [C:\publishaot2\publishaot2.csproj]
```

The ILCompiler targets set PublishTrimmed to true, but I assume they get included too late from the SDK to affect feature switches. We should do this propagation very early.

If I instead publish as `dotnet publish -r win-x64 -c Release /p:PublishAot=true /p:PublishTrimmed=true` there's no warning, but I shouldn't have to do that.

Cc @dotnet/ilc-contrib

### PR DESCRIPTION
**Pull request by LakshanF**
Fixes https://github.com/dotnet/sdk/issues/25740

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 1]
```diff
@@ -46,7 +46,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <!-- Set up default feature switches -->
   <PropertyGroup>
-    <EnableUnsafeBinaryFormatterSerialization Condition="'$(EnableUnsafeBinaryFormatterSerialization)' == ''">false</EnableUnsafeBinaryFormatterSerialization>
+    <CustomResourceTypesSupport Condition="'$(CustomResourceTypesSupport)' == ''">false</CustomResourceTypesSupport>
```
Comment (jkotas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:49]: We do want to have `EnableUnsafeBinaryFormatterSerialization` to be off by default in NativeAOT. I think you want to keep the existing condition and add the condition for `CustomResourceTypesSupport` in addition to that.

**Review by MichalStrehovsky — CHANGES_REQUESTED** [review]
This is set in Microsoft.NET.ILLink.targets:

https://github.com/dotnet/sdk/blob/b15a88fb330548bf447d2df4936c4367db346a1a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L38

The default is always false when trimming. We do not want to be duplicating trimming defaults in these targets.

I think the actual problem here (and the reason why the warning disappears if one sets `PublishTrimmed` to true in the project file) is that ILCompiler.targets (that set PublishTrimmed to true) are included after ILLink.targets. Can we reorder them (or do something else that makes sure PublishTrimmed is set when PublishAot is set)?

https://github.com/dotnet/sdk/blob/ad1ca90ad286288970a4ae1ec77e85848fef5a24/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L1157-L1158

**Comment by MichalStrehovsky** [comment]
(Or maybe we need an ILCompiler.props file that sets the default for PublishTrimmed so that all the targets already see it as set?)

**Comment by LakshanF** [comment]
Fixed the warnings in the SDK, https://github.com/dotnet/sdk/pull/25927, as suggested by @MichalStrehovsky above

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets (hunk 1 above)
Comment (LakshanF) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets]: Both these (old and new) feature switches are disabled in [Microsoft.NET.ILLink.targets](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L37-L39) now when `PublishTrimmed `is set to true. That now happens with the SDK fix, https://github.com/dotnet/sdk/pull/25927

**Review by MichalStrehovsky — APPROVED** [review]

**Pull request merged**